hack/devtools/rp_dev_helper.sh

```diff
@@ -1,39 +1,471 @@
-get_digest_tag() {
-    local IMAGE_NAME=$1
-    local IMAGE_TAG=$(extract_image_tag "$IMAGE_NAME" "$VERSION_CONST_FILE")
-    echo "$IMAGE_NAME and Tag: $IMAGE_TAG"
-    echo "$IMAGE_TAG"
+check_acr_repo() {
+    err_str="Usage: $0 <ACR_NAME> <REPOSITORY> [SKIP_DEPLOYMENTS] [TAG]. Please try again"
+    local acr_name=${1?$err_str}
+    local repository=${2?$err_str}
+    local skip_deployments=${3:-"true"}
+    local tag=${4:-"no-tag"}
+
+   # Don't skip deployment creation when skip_deployments was set to 'false'
+    if is_it_boolean $skip_deployments && ! $skip_deployments; then
+        log "'skip_deployments' was set to 'false'. ❌⏩ Don't skip ACR '$acr_name' repo mirroring for repository '${repository}'."
+        return 1
+    fi
+
+    # Check if the repository tag is not empty and if it matches an optional tag
+    repo_tag=$(az acr repository show-tags --name "$acr_name" --repository "${repository}" -o tsv | tr '' ' ')
+    if [[ -n "$repo_tag" ]]; then
+        # TODO: Loop all the repo tags and print all the tags in one line
+        if [[ "${tag}" != "no-tag" && "${tag}" != "${repo_tag}" ]] ; then
+            log "🔴✈️ Repository '${repository}' in ACR '$acr_name' exists, but with a wrong tag '${repo_tag}'. Expected tag: '${tag}'."
+            return 1
+        fi
+        log "🟢✈️ Repository '${repository}' in ACR '$acr_name' exists with tag '${repo_tag}'."
+        return 0
+    fi
+    log "🔴❌✈️ Repository '${repository}' doesn't exist in ACR '$acr_name'."
+    return 1
+}
+
+# "openshift-release-dev/ocp-release" "openshift-release-dev/ocp-v4.0-art-dev" were excluded as they don't include an image/tag
+acr_repos=("azure-cli" "rhel8/support-tools" "rhel9/support-tools" "openshift4/ose-tools-rhel8" "ubi8/ubi-minimal" "ubi9/ubi-minimal"
+        "ubi8/nodejs-18" "ubi8/go-toolset" "app-sre/managed-upgrade-operator" "app-sre/hive" "distroless/genevamdm" "distroless/genevamdsd"
+        "aro" "fluentbit")
+
+check_acr_repos() {
+    err_str="Usage: $0 <RESOURCE_GROUP> [GIT_COMMIT] [SKIP_DEPLOYMENTS]. Please try again"
+    local resource_group=${1?$err_str}
+    local git_commit=${2:-"no-commit"}
+    local skip_deployments=${3:-"true"}
+
+    # Don't skip deployment creation when skip_deployments was set to 'false'
+    if is_it_boolean $skip_deployments && ! $skip_deployments; then
+        log "'skip_deployments' was set to 'false'. ❌⏩ Don't skip acr repo mirroring in ResourceGroup $resource_group"
+        return 1
+    fi
+   
+    # check_jq_installed - Might not needed
+    # Get the first Azure Container Registry (ACR) name under ResourceGroup
+    acr_name="$(az acr list --resource-group $resource_group | jq -r '.[0].name')"
+    if [[ -z "$acr_name" ]]; then
+        abort "🔴❌✈️ Error: There are no Azure Container Registries under ResourceGroup '$resource_group'."
+    fi
+    
+    # Do all the needed repos already imported
+    local -a missing_repos_names=()
+    for repo in ${acr_repos[@]}; do
```

Comment:
Add shellcheck disable comment for shellcheck linter.

```suggestion
    # shellcheck disable=SC2068
    for repo in ${acr_repos[@]}; do
```